jsonschema/src/keywords/contains.rs

```diff
@@ -433,19 +433,34 @@ pub(crate) fn compile<'a>(
                                 max_contains,
                             ))
                         } else {
-                            Some(Err(ValidationError::schema(schema)))
+                            Some(Err(ValidationError::format(
```

Comment:
Here we can check whether the value is `i64` - if so, then it is a negative value (otherwise it should be castable to `u64` in the code above). In this case, `ValidationError::minimum`  should cover our need, otherwise - `ValidationError::single_type_error`. 

This comment should be applicable to other cases in this file & similar cases for `maxItems`, `maxLength`, `maxProperties`, `minItems`, `minLength`, and `minProperties`

Reply:
> Here we can check whether the value is i64 - if so, then it is a negative value

Its not necessarily true, e.g `1` is a valid `i64` but its not negative. So it would require `< 0` check.

Reply:
If it is `1`, then the execution will go in the branch above, but if it is in this branch, then we exclude all values that are `u64` AND `i64`, so only negative values are possible

Reply:
Check how I did implement it.

https://github.com/Stranger6667/jsonschema-rs/pull/280/files#diff-96d9e0c98ab5e4b39ff4c741e1ab1958b48e5aa4688f4e0d42e420a2cc2dedabR449-R474